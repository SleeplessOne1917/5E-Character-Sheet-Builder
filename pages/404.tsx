import Four04View from '../src/views/404/404';
import { NextPage } from 'next';

const Four04Page: NextPage = () => <Four04View />;

export default Four04Page;
