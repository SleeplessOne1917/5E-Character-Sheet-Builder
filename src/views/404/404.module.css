.content {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 60%;
    flex-direction: column;
}

.text {
    font-family: var(--font-fantasy);
    color: var(--davy);
    text-align: center;
}

.four04 {
    font-size: 3rem;
    font-weight: bold;
}

.not-found {
    padding: 0.5rem;
    font-size: 2rem;
}

.blurb {
    font-size: 1.5rem;
    font-family: var(--font-display) !important;
    padding: 1.5rem;
}

.countdown {
    font-weight: bold;
    color: var(--wine-red);
}