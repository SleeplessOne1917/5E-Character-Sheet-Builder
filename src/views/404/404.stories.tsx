import { ComponentMeta, ComponentStory } from '@storybook/react';

import Four04 from './404';

export default {
	title: 'Views/404',
	component: Four04
} as ComponentMeta<typeof Four04>;

const Template: ComponentStory<typeof Four04> = () => <Four04 />;

export const Default = Template.bind({});
