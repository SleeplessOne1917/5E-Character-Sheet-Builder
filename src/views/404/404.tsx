import MainContent from '../../components/MainContent/MainContent';
import classes from './404.module.css';
import { useEffect } from 'react';
import useRedirectCountdown from '../../hooks/useRedirectCountdown';

const Four04 = () => {
	const { secondsLeft, startCountdown } = useRedirectCountdown({
		path: '/',
		replace: true,
		seconds: 10
	});

	useEffect(() => {
		startCountdown();
	}, [startCountdown]);

	return (
		<MainContent testId="404">
			<div className={classes.content}>
				<div className={`${classes.text} ${classes.four04}`}>404</div>
				<div className={`${classes.text} ${classes['not-found']}`}>
					Page not found.
				</div>
				<div className={`${classes.text} ${classes.blurb}`}>
					Redirecting to the home page in{' '}
					<span className={classes.countdown}>
						{secondsLeft} second{secondsLeft === 1 ? '' : 's'}
					</span>
					.
				</div>
			</div>
		</MainContent>
	);
};

export default Four04;
