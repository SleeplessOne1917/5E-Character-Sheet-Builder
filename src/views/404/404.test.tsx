import * as stories from './404.stories';

import { render, screen } from '@testing-library/react';

import { composeStories } from '@storybook/testing-react';

const { Default } = composeStories(stories);

it('renders correctly', () => {
	render(<Default />);

	expect(screen.getByTestId('404')).toMatchSnapshot();
});
