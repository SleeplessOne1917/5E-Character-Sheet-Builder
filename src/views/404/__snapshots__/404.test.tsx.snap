// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  class="main"
  data-testid="404"
>
  <div
    class="content"
  >
    <div
      class="content"
    >
      <div
        class="text four04"
      >
        404
      </div>
      <div
        class="text not-found"
      >
        Page not found.
      </div>
      <div
        class="text blurb"
      >
        Redirecting to the home page in
         
        <span
          class="countdown"
        >
          10
           second
          s
        </span>
        .
      </div>
    </div>
  </div>
</main>
`;
